File: java/bundles/org.eclipse.set.model.siteplan/src/org/eclipse/set/model/siteplan/Position.java
/**
 * Copyright (c) 2021 DB Netz AG and others.
 * 
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 */
package org.eclipse.set.model.siteplan;

/**
 * <!-- begin-user-doc --> A representation of the model object
 * '<em><b>Position</b></em>'. <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link org.eclipse.set.model.siteplan.Position#getRotation <em>Rotation</em>}</li>
 * </ul>
 *
 * @see org.eclipse.set.model.siteplan.SiteplanPackage#getPosition()
 * @model
 * @generated
 */
public interface Position extends Coordinate {
	/**
	 * Returns the value of the '<em><b>Rotation</b></em>' attribute.
	 * The default value is <code>"0"</code>.
	 * <!-- begin-user-doc --> <!--
	 * end-user-doc -->
	 * @return the value of the '<em>Rotation</em>' attribute.
	 * @see #setRotation(double)
	 * @see org.eclipse.set.model.siteplan.SiteplanPackage#getPosition_Rotation()
	 * @model default="0"
	 * @generated
	 */
	double getRotation();

	/**
	 * Sets the value of the
	 * '{@link org.eclipse.set.model.siteplan.Position#getRotation
	 * <em>Rotation</em>}' attribute. <!-- begin-user-doc --> <!-- end-user-doc
	 * -->
	 * 
	 * @param value
	 *            the new value of the '<em>Rotation</em>' attribute.
	 * @see #getRotation()
	 * @generated
	 */
	void setRotation(double value);

} // Position

File: web/siteplan/src/model/Position.ts
import { checkInstance } from '@/util/ObjectExtension'

/**
 * Copyright (c) 2022 DB Netz AG and others.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 */
export enum DBRef {
  CR0 = 'CR0',
  DR0 = 'DR0',
  ER0 = 'ER0',
  FR0 = 'FR0'
}
export interface Coordinate {
    x: number
    y: number
}

export function defaultCoordinateObj () : Coordinate {
  return {
    x: 0,
    y: 0
  }
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isInstanceOfCoordinate (object: any): boolean {
  return checkInstance(object, defaultCoordinateObj())
}

export interface Position extends Coordinate {
    rotation: number
}

export function defaultPositionObj () : Position {
  return {
    x: 0,
    y: 0,
    rotation: 0
  }
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function isInstanceOfPosition (object: any): boolean {
  return checkInstance(object, defaultPositionObj())
}
